commands/cli/parse.go

```diff
@@ -75,67 +58,142 @@ func Parse(input []string, stdin *os.File, root *cmds.Command) (cmds.Request, *c
 	return req, cmd, path, nil
 }
 
-// parsePath separates the command path and the opts and args from a command string
-// returns command path slice, rest slice, and the corresponding *cmd.Command
-func parsePath(input []string, root *cmds.Command) ([]string, []string, *cmds.Command) {
-	cmd := root
-	path := make([]string, 0, len(input))
-	input2 := make([]string, 0, len(input))
+// Parse a command line made up of sub-commands, short arguments, long arguments and positional arguments
+func parseOpts(args []string, root *cmds.Command) (
+	path []string,
+	opts map[string]interface{},
+	stringVals []string,
+	cmd *cmds.Command,
+	err error,
+) {
+	path = make([]string, 0, len(args))
+	stringVals = make([]string, 0, len(args))
+	optDefs := map[string]cmds.Option{}
+	opts = map[string]interface{}{}
+	cmd = root
+
+	// parseFlag checks that a flag is valid and saves it into opts
+	// Returns true if the optional second argument is used
+	parseFlag := func(name string, arg *string, mustUse bool) (bool, error) {
```

Comment:
Thanks for reviewing. Yes, this argument can be nil. Is there a more idiomatic way to represent optional arguments?